README.md

```diff
@@ -1,173 +1,175 @@
+
+### Account queries
+
+The current balance of any account can be queried using `Erdstall.getAccount(who)`.
+
+### Entering and leaving
+
+Fungible Tokens and NFTs can be deposited into the Erdstall network by calling
+`ErdstallSession.deposit(assets)`. This call returns a list of promises of on-chain transactions. Only after all those transactions are mined are the deposits successful. The assets will be available in Erdstall once the blocks containing them have reached the enclave and sufficient finality depth has been reached.
+
+To exit an Erdstall network with all assets, call `ErdstallSession.leave()`. This will first send an `exit` request to the Erdstall enclave. Once the epoch has ended, an exit proof will be received from the operator. Due to protocol-security-related reasons, it now has to be waited for another full epoch before an on-chain `withdraw` transaction can be sent to the Erdstall contract, containing the exit proof that was received in the prior epoch. The call returns a list of transaction promises. Once all those transactions have been mined, the user's assets will be withdrawn from Erdstall back into their Ethereum wallet.
+
+### Erdstall transactions
+
+The following transactions are currently possible in any Erdstall network.
+
+#### Transfers
+
+To transfer assets within Erdstall, use `Erdstall.transferTo(assets, to)`.
+This transfers the specified assets to the specified recipient, and returns a transaction receipt.
+
+#### Trading
+
+Erdstall natively supports (basket) trading of tokens and NFTs. Currently, simple two-step offer and accept trading is supported. Some external marketplace infrastructure needs to be in place to present trade offers to potential buyers, who can then execute a trade on the Erdstall network.
+
+A trade offer can be created using `ErdstallSession.createOffer()`. The returned `TradeOffer` needs the external marketplace infrastructure to reach potential buyers. An interested buyer can then execute the trade by calling `ErdstallSession.acceptTrade(offer)`, which will send a `Trade` transaction to the Erdstall network, atomically exchanging the assets between the two accounts.
+
+#### NFT minting & burning
+
+`ErdstallSession.mint(token, id)` can be used to mint a new NFT with on-chain contract address `token` and id `id`. The NFT can then be traded freely and is only minted on-chain upon withdrawal.
+
+NFT-owners can burn their tokens with a `ErdstallSession.burn(nfts)` transaction call. If the NFT was deposited into Erdstall, it will not be burned on-chain, but will indefinitely be locked-up and be held by the NFT-Holder contract.
+
+## Event Subscriptions & Callbacks
+
+After creating an Erdstall client, event handlers can be set up using `Erdstall.on()` and `Erdstall.once()`, like
```

Comment:
What is the difference, is there a difference? Maybe some more introduction?

Reply:
Yes there is a difference. `on` handlers are persistent and are called repeatedly, whereas `once` handlers are only called once and then removed. This is a common JS API which JS devs usually know. I'll add a sentence nonetheless.